File: Rust/project_one/src/part.rs

pub struct Part {
    pub part_name: String,
    pub part_number: i64,
    pub list_price: f64,
    pub sale_price: f64,
    pub on_sale: bool,
    pub quantity: i64
}
impl Part {
    /*fn create (part_name: str, part_number: i64, list_price: f64, sale_price: f64, on_sale: bool, quantity: i64) -> Part {
        Part{partname: part_name, part_number: part_number, list_price: list_price, sale_price: sale_price, on_sale: on_sale, quantity: quantity}
    }*/
    /*fn as_string_record (&self) -> StringRecord {
        let mut record = StringRecord::new();
        record.push_field(&self.part_name);
        record.push_field(&self.part_number);
        record.push_field(&self.list_price);
        record.push_field(&self.sale_price);
        record.push_field(&self.on_sale);
        record.push_field(&self.quantity);
        return record;
    }*/
}

File: java/Part.java
public class Part {
  private String partName;
  private Long partNumber;
  private Double listPrice;
  private Double salePrice;
  private boolean onSale;
  private Long quantity;

  public Part(String partName, Long partNumber, Double listPrice, Double salePrice, boolean onSale, Long quantity) {
    this.partName = partName;
    this.partNumber = partNumber;
    this.listPrice = listPrice;
    this.salePrice = salePrice;
    this.onSale = onSale;
    this.quantity = quantity;
  }

  public String getPartName() {
    return this.partName;
  }

  public Long getPartNumber() {
    return this.partNumber;
  }
  public Double getListPrice() {
    return this.listPrice;
  }
  public Double getSalePrice() {
    return this.salePrice;
  }
  public boolean isOnSale() {
    return this.onSale;
  }
  public Long getQuantity() {
    return this.quantity;
  }

  public void setPartName(String partName) {
    this.partName = partName;
  }

  public void setPartNumber(Long partNumber) {
    this.partNumber = partNumber;
  }
  public void setListPrice(Double listPrice) {
    this.listPrice = listPrice;
  }
  public void setSalePrice(Double salePrice) {
    this.salePrice = salePrice;
  }
  public void setOnSale(boolean onSale) {
    this.onSale = onSale;
  }
  public void setQuantity(Long quantity) {
    this.quantity = quantity;
  }

  public void decrement() {
    this.quantity--;
  }

  public void decrement(int amt) {
    this.quantity = this.quantity - amt;
  }

  public void increment() {
    this.quantity++;
  }

  public void increment(int amt) {
    this.quantity = this.quantity + amt;
  }

}
